frontend/src/lib/components/IntervalFilter/IntervalFilter.tsx

```diff
@@ -1,19 +1,40 @@
 import { intervalFilterLogic } from './intervalFilterLogic'
 import { useActions, useValues } from 'kea'
 import { intervals } from 'lib/components/IntervalFilter/intervals'
-import { InsightType, IntervalType } from '~/types'
+import { BaseMathType, InsightType, IntervalType } from '~/types'
 import { insightLogic } from 'scenes/insights/insightLogic'
-import { LemonSelect } from '@posthog/lemon-ui'
+import { LemonSelect, lemonToast } from '@posthog/lemon-ui'
 
 interface InvertalFilterProps {
     view: InsightType
     disabled?: boolean
+    activeUsersMath?: BaseMathType.WeeklyActiveUsers | BaseMathType.MonthlyActiveUsers | null
 }
 
-export function IntervalFilter({ disabled }: InvertalFilterProps): JSX.Element {
+export function IntervalFilter({ disabled, activeUsersMath }: InvertalFilterProps): JSX.Element {
     const { insightProps } = useValues(insightLogic)
     const { interval } = useValues(intervalFilterLogic(insightProps))
     const { setInterval } = useActions(intervalFilterLogic(insightProps))
+
+    const enabledIntervals: Record<string, any> = { ...intervals }
+
+    if (activeUsersMath) {
+        // Disallow grouping by hour for WAUs/MAUs as it's an expensive query that produces a view that's not useful for users
+        delete enabledIntervals.hour
+
+        // Disallow grouping by month for WAUs as the resulting view is misleading to users
+        if (activeUsersMath === BaseMathType.WeeklyActiveUsers) {
+            delete enabledIntervals.month
+        }
+    }
+
+    if (interval && !enabledIntervals[interval]) {
+        const humanReadableMathType =
+            activeUsersMath === BaseMathType.MonthlyActiveUsers ? 'Monthly active users' : 'Weekly active users'
+        lemonToast.info(`${humanReadableMathType} does not support grouping by ${interval}. Grouping by day instead.`)
```

Comment:
If you're switching from month-by-month MAU to WAU, switching to day-by-day doesn't spark a lot of joy. But if we knew to switch to week-by-week in that case… That'd be perfectly smooth